core/teverseUI/console.lua

```diff
@@ -1,75 +1,90 @@
-    visible = false
+	parent = teverse.coreInterface,
+	size = guiCoord(0,
+		math.max(teverse.coreInterface.absoluteSize.x / 3, 100),
+		0,
+		math.max(teverse.coreInterface.absoluteSize.y / 3, 200));
+	position = guiCoord(0, 20, 0, 20);
+	visible = false;
+	strokeWidth = 2;
+	strokeColour = colour(0.75,0.75,0.75);
+	strokeAlpha = 1;
+	strokeRadius = 2;
+})
+
+local topBar = teverse.construct("guiFrame", {
+	parent = container;
+	size = guiCoord(1, 0, 0, 20);
+	backgroundColour = colour(0.75, 0.75, 0.75);
 })
 
-local top = teverse.construct("guiTextBox", {
-    parent = container,
-    size = guiCoord(1, 0, 0, 20),
-    position = guiCoord(0, 0, 0, 0),
-    backgroundAlpha = 0.5,
-    textSize = 20,
-    textAlign = "middleLeft",
-    -- bit hacky but works for slight indent
-    text = " Console",
-    backgroundColour = colour.rgb(112, 112, 112)
+local title = teverse.construct("guiTextBox", {
+	parent = topBar;
+	size = guiCoord(1, -24, 1, 0);
+	position = guiCoord(0, 4, 0, 0);
+	backgroundAlpha = 0;
+	active = false;
+	textSize = 20;
+	text = "Console"
+})
 
+local leave = teverse.construct("guiFrame", {
+	parent = topBar;
+	size = guiCoord(0, 14, 0, 14);
+	position = guiCoord(1, -16, 0, 2);
+	backgroundColour = colour(0.70, 0.35, 0.35);
+	strokeRadius = 14
 })
 
-local logContainer = teverse.construct("guiScrollView", {
-    parent = container,
-    size = guiCoord(1, -10, 1, -27),
-    position = guiCoord(0, 5, 0, 20),
-    backgroundAlpha = 0.0,
-    canvasSize = guiCoord(1, -1, 10, 0),
-    scrollbarWidth = 3,
-    scrollbarRadius = 2
+local clear = teverse.construct("guiTextBox", {
+	parent = topBar;
+	text = "C";
+	textColour =  colour(0.5, 0.5, 0.5);
+	textAlign = "middle";
+	textSize = 15;
+	size = guiCoord(0, 14, 0, 14);
+	position = guiCoord(1, -32, 0, 2);
+	backgroundColour = colour(0.8, 0.8, 0.8);
+	strokeRadius = 14
 })
 
+local commandLine = require("tevgit:core/teverseUI/commandLine.lua")
+commandLine.parent = container
+commandLine.size = guiCoord(1, 0, 0, 20)
+commandLine.position = guiCoord(0, 0, 1, -20)
 
-draggableUi(container, top)
-local lastPos = 0
-function addLog (msg, time)
-    local txt = teverse.construct("guiTextBox", {
-        parent = logContainer,
-        size = guiCoord(1, -10, 0, 25),
-        position = guiCoord(0, 5, 0, lastPos),
-        backgroundAlpha = 0,
-        strokeAlpha = 0,
-        textWrap = true,
-        clip = false
-    })
-    txt.text = os.date("%H:%M:%S", time) .. " : " .. msg
-    txt.size = guiCoord(1, -10, 0, txt.textDimensions.y)
-    lastPos = lastPos + txt.textDimensions.y + 3
+local log = require("tevgit:core/teverseUI/log.lua")({
+	parent = container;
+	size = guiCoord(1, 0, 1, -40);
+	position = guiCoord(0, 0, 0, 20);
+	scrollbarRadius = 0.01;
+	scrollbarWidth = 4;
+	scrollbarColour = colour(.8,.8,.8);
+})
 
-    if string.find(msg, "ERROR:") then
-        txt.textColour = colour.rgb(220, 50, 47)
-    end
+--Allow users to move and resize the console
```

Comment:
Is it possible to require the functions separately before executing them? Purely for readability
i.e.
```lua
-- at top of file
local makeResizable = require("...")
```

Reply:
That is a good point, though I decided on using it where needed because I use them once. There isn't much of a point listing on the top if its purpose is clear at require.